Doesn't compile. Remove the ``|| charflag``

```SourceX/miniwin/misc_msg.cpp: In function 'dvl::WINBOOL dvl::PeekMessageA(dvl::LPMSG, dvl::HWND, dvl::UINT, dvl::UINT, dvl::UINT)':
SourceX/miniwin/misc_msg.cpp:193:20: error: 'charflag' was not declared in this scope
     if (invflag || charflag) {
                    ^~~~~~~~
SourceX/miniwin/misc_msg.cpp:193:20: note: suggested alternative: 'chrflag'
     if (invflag || charflag) {
                    ^~~~~~~~
                    chrflag
make: *** [makefile:302: obj/misc_msg.o] Error 1

--- a/README.md
+++ b/README.md
@@ -13,12 +13,14 @@
 - Left Analog : Move Hero
 - Right Analog : Simulate Mouse
 - B : Attack nearby enemies, talk to towns people and merchants. Pickup & Drop items in inventory.
-- Y : Pickup gold, potions & equipment from ground, open chests and doors that are nearby.
+- Y : Pickup gold, potions & equipment from ground, open chests and doors that are nearby. Use item when in inventory (read books etc.).
 - X : Cast Spell, Previous Menu
 - A : Inventory
 - R : Character
 - ZR : Drink Mana Potion
 - ZL : Drink Heal Potion
+- Left Analog Click : Quest Log
+- Right Analog Click : Perform left mouse click action on selected object or monster
 - Minus : Automap
 - Plus : Game Menu
 

--- a/Source/diablo.cpp
+++ b/Source/diablo.cpp
@@ -1369,21 +1369,24 @@ void PressChar(int vkey)
 			}
 
 			if(!chrflag || invflag) {
+				if(MouseX > 160 && MouseY < VIEWPORT_HEIGHT) {
+					SetCursorPos(MouseX - 160, MouseY);
+					MouseX = MouseX - 160;
+					MouseY = MouseY;
+				}
+			} else {
 				if (!chrbtnactive && plr[myplr]._pStatPts) {
 					int x = attribute_inc_rects2[0][0] + (attribute_inc_rects2[0][2] / 2);
 					int y = attribute_inc_rects2[0][1] + (attribute_inc_rects2[0][3] / 2);
 					SetCursorPos(x, y);
 					MouseX = x;
 					MouseY = y;
-				}
-				/*if(MouseX > 160 && MouseY < VIEWPORT_HEIGHT) {
-					SetCursorPos(MouseX - 160, MouseY);
-				}*/
-			} else {
-				if(MouseX < 480 && MouseY < VIEWPORT_HEIGHT) {
-					SetCursorPos(MouseX + 160, MouseY);
-					MouseX = MouseX + 160;
-					MouseY = MouseY;
+				} else {
+					if(MouseX < 480 && MouseY < VIEWPORT_HEIGHT) {
+						SetCursorPos(MouseX + 160, MouseY);
+						MouseX = MouseX + 160;
+						MouseY = MouseY;
+					}
 				}
 			}
 		}

--- a/Source/plrctrls.cpp
+++ b/Source/plrctrls.cpp
@@ -451,7 +451,7 @@ void hotSpellMove(int key)
  
 void walkInDir(int dir)
 {
-	if (invflag || spselflag || chrflag) // don't walk if inventory, speedbook or char info windows are open
+	if (invflag || spselflag || chrflag || questlog) // don't walk if inventory, speedbook or char info windows are open
 		return;
 	ticks = GetTickCount();
 	if (ticks - invmove < 370) {

--- a/Source/plrctrls.cpp
+++ b/Source/plrctrls.cpp
@@ -571,16 +571,21 @@ void keyboardExpansion(int vikey)
 		} else if (chrflag) {
 			if (ticks - statuptimer >= 400) {
 				statuptimer = ticks;
-				for (int i = 0; i < 4; i++) {
-					if (MouseX >= attribute_inc_rects2[i][0]
-					    && MouseX <= attribute_inc_rects2[i][0] + attribute_inc_rects2[i][2]
-					    && MouseY >= attribute_inc_rects2[i][1]
-					    && MouseY <= attribute_inc_rects2[i][3] + attribute_inc_rects2[i][1]) {
-						chrbtn[i] = 1;
-						chrbtnactive = TRUE;
-						ReleaseChrBtns();
+				if (!chrbtnactive && plr[myplr]._pStatPts) {
+					CheckChrBtns();
+					for (int i = 0; i < 4; i++) {
+						if (MouseX >= attribute_inc_rects2[i][0]
+							&& MouseX <= attribute_inc_rects2[i][0] + attribute_inc_rects2[i][2]
+							&& MouseY >= attribute_inc_rects2[i][1]
+							&& MouseY <= attribute_inc_rects2[i][3] + attribute_inc_rects2[i][1]) {
+							chrbtn[i] = 1;
+							chrbtnactive = TRUE;
+							ReleaseChrBtns();
+						}
 					}
 				}
+				if (plr[myplr]._pStatPts == 0)
+					HideCursor();
 			}
 		} else {
 			HideCursor();

--- a/SourceX/miniwin/misc_msg.cpp
+++ b/SourceX/miniwin/misc_msg.cpp
@@ -301,6 +301,20 @@ WINBOOL PeekMessageA(LPMSG lpMsg, HWND hWnd, UINT wMsgFilterMin, UINT wMsgFilter
 		}
 		#endif
 		break;
+	case SDL_JOYBUTTONUP:
+		#if defined(SWITCH)
+		switch(e.jbutton.button)
+		{
+			case  3:	// Y
+				if (invflag) {
+					lpMsg->message = DVL_WM_RBUTTONUP;
+					lpMsg->lParam = (MouseY << 16) | (MouseX & 0xFFFF);
+					lpMsg->wParam = keystate_for_mouse(0);
+				}
+				break;
+		}
+		#endif
+		break;
 	case SDL_QUIT:
 		lpMsg->message = DVL_WM_QUIT;
 		break;

--- a/SourceX/miniwin/misc_msg.cpp
+++ b/SourceX/miniwin/misc_msg.cpp
@@ -190,8 +190,17 @@ WINBOOL PeekMessageA(LPMSG lpMsg, HWND hWnd, UINT wMsgFilterMin, UINT wMsgFilter
 				PressChar('x');
 				break;
 			case  3:	// Y
-				PressKey(VK_RETURN);
-				keyboardExpansion(VK_RETURN);
+				if (invflag) {
+					lpMsg->message = DVL_WM_RBUTTONDOWN;
+					lpMsg->lParam = (MouseY << 16) | (MouseX & 0xFFFF);
+					lpMsg->wParam = keystate_for_mouse(DVL_MK_RBUTTON);
+				} else {
+					PressKey(VK_RETURN);
+					keyboardExpansion(VK_RETURN);
+				}
+				break;
+			case  4:	// left joystick click
+				PressChar('q');
 				break;
 			case  5:	// right joystick click
 				if (newCurHidden) { // show cursor first, before clicking

